umap/views.py

```diff
@@ -778,6 +779,36 @@ def form_valid(self, form):
         return response
 
 
+def get_websocket_auth_token(request, map_id, map_inst):
+    """Return an signed authentication token for the currently
+    connected user, allowing edits for this map over WebSocket.
+
+    If the user is anonymous, return a signed token with the map id.
+
+    The returned token is a signed object with the following keys:
+    - user: user primary key OR "anonymous"
+    - map_id: the map id
+    - permissions: a list of allowed permissions for this user and this map
+    """
+    map_object: Map = Map.objects.get(pk=map_id)
+
+    if map_object.can_edit(request.user, request):
+        permissions = ["edit"]
+        if map_object.is_owner(request.user, request):
+            permissions.append("owner")
+
+        if request.user.is_authenticated:
+            user = request.user.pk
+        else:
+            user = "anonymous"
+        signed_token = TimestampSigner().sign_object(
+            {"user": user, "map_id": map_id, "permissions": permissions}
+        )
+        return simple_json_response(token=signed_token)
+    else:
+        return HttpResponseForbidden
```

Comment:
```suggestion
        return HttpResponseForbidden(_("You cannot edit this map with your current permissions."))
```

You need to return an instance here (even better with a dedicated message).

Reply:
Indeed 🤦